apis_bibsonomy/models.py

```diff
@@ -29,3 +32,25 @@ def bibtexjson(self):
 
     def get_absolute_url(self):
         return reverse("apis_bibsonomy:referencedetail", kwargs={"pk": self.pk})
+
+    def save(self):
+        bib_ref = self.bibs_url
+        sources = getattr(settings, "APIS_BIBSONOMY", [])
+        sources = [s for s in sources if s.get("url") and s.get("url") in self.bibs_url]
+
+        source = sources[-1]
+
+        if self.bibs_url is not None:
+            btype = source.get("type", None)
+            if btype == "bibsonomy":
+                bib_e = BibsonomyEntry(bib_hash=bib_ref, base_set=sett1)
+                self.bibtex = json.dumps(bib_e.bibtex)
+            elif btype == "zotero":
+                headers = {
+                    "Zotero-API-Key": source.get("API key", None),
+                    "Zotero-API-Version": "3",
+                }
+                params = {"include": "csljson"}
+                res = requests.get(bib_ref, headers=headers, params=params)
+                self.bibtex = json.dumps(res.json()["csljson"])
+        super().save()
```

Comment:
Shouldn't this line be indented to only save the object if the object actually has a bibs-ref and therefore a referenced object?  And / or raise an error if bibs-ref is missing?